contrib/transaction_profiling_analyzer/transaction_profiling_analyzer.py

```diff
@@ -508,155 +514,199 @@ def has_dateparser():
-            if start_range_left < key:
-                if end_key <= key:
-                    self.ranges[start_range_left] = end_key, 1
-                    return
-                self.ranges[start_range_left] = key, 1
-                start_range_left = key
-            assert start_range_left >= key
-            if start_range_left >= cur_end_key:
-                continue
-
-            # [key, start_range_left) = cur_count
-            # if key == start_range_left this will get overwritten below
-            self.ranges[key] = start_range_left, cur_count
-
-            if end_key <= cur_end_key:
-                # [start_range_left, end_key) = cur_count+1
-                # [end_key, cur_end_key) = cur_count
-                self.ranges[start_range_left] = end_key, cur_count + 1
-                if end_key != cur_end_key:
-                    self.ranges[end_key] = cur_end_key, cur_count
-                start_range_left = end_key
-                break
-            else:
-                # [start_range_left, cur_end_key) = cur_count+1
-                self.ranges[start_range_left] = cur_end_key, cur_count+1
-                start_range_left = cur_end_key
-            assert start_range_left <= end_key
-
-        # there may be some range left
-        if start_range_left < end_key:
-            self.ranges[start_range_left] = end_key, 1
-
-    def get_count_for_key(self, key):
-        if key in self.ranges:
-            return self.ranges[key][1]
-
-        keys = self.ranges.keys()
-        index = bisect_left(keys, key)
-        if index == 0:
-            return 0
-
-        index_key = keys[index-1]
-        if index_key <= key < self.ranges[index_key][0]:
-            return self.ranges[index_key][1]
-        return 0
-
-    def get_range_boundaries(self, shard_finder=None):
-        total = sum([count for _, (_, count) in self.ranges.items()])
-        range_size = total // self.k
+            self._insert_read(get_range.key_range.start_key, get_range.key_range.end_key)
+
+    def _insert_read(self, start_key, end_key):
+        self.read_counts.setdefault((start_key, end_key), 0)
+        self.read_counts[(start_key, end_key)] += 1
+
+        self.reads.setdefault(start_key, [0, 0])[0] += 1
+        if end_key is not None:
+            self.reads.setdefault(end_key, [0, 0])[1] += 1
+        else:
+            self.reads.setdefault(start_key+b'\x00', [0, 0])[1] += 1
+
+    def get_total_reads(self):
+        return sum([v for v in self.read_counts.values()])
+    
+    def matches_filter(addresses, required_addresses):
+        for addr in required_addresses:
+            if addr not in addresses:
+                return False
+        return True
+
+    def get_top_k_reads(self, num, filter_addresses, shard_finder=None):
+        count_pairs = sorted([(v, k) for (k, v) in self.read_counts.items()], reverse=True, key=lambda item: item[0])
```

Comment:
I honestly don't remember at this point. I have some vague memory of resolving an issue by doing this, but it could have been when the code looked a little different.

Reply:
Oh I think I remember, the problem is that the values for `k` can be of different types that aren't comparable, so I had to write this code to avoid comparing them.

Reply:
Isn't `k` always bytes ?

Reply:
K is tuple that could either contain two `bytes` strings for a range read, or 1 byte string and a `None` for point reads.

Reply:
Yeah using empty byte string in place of `None` would solve the problem?

Reply:
It's slightly ambiguous, since you could do a range read from `''` to `''`, but otherwise it could work. But the above solution also works fine, we don't need to be able to compare these key tuples.